tests/test_pattoo_shared/test_daemon.py

```diff
@@ -34,46 +103,211 @@ class TestDaemon(unittest.TestCase):
     # General object setup
     #########################################################################
 
+    def setUp(self):
+        """Test setup"""
+
+        # Setup base config and agent
+        self._agent = create_agent()
+        self._config = _config
+
+        # Instantiation of test daemon
+        self._daemon = MockDaemon(self._agent)
+
+    def tearDown(self):
+        """Test clean up"""
+        self._config = None
+        self._agent = None
+
+        # Removing any relate lock file
+        if os.path.exists(self._daemon.lockfile):
+            os.remove(self._daemon.lockfile)
+        if os.path.exists(self._daemon.pidfile):
+            os.remove(self._daemon.pidfile)
+
+        self._daemon = None
+
     def test___init__(self):
         """Testing function __init__."""
-        pass
+        # Check daemon name matches agent name
+        self.assertEqual(self._daemon.name, AGENT_NAME)
+
+        # Checking daemon config
+        self.assertEqual(self._daemon._config, self._config)
+        self.assertEqual(self._daemon._config, self._config)
+
+        # Checking daemon pid_file
+        expected = files.pid_file(AGENT_NAME, self._config)
+        self.assertEqual(self._daemon.pidfile, expected)
+
+        # Checking daemon lock_file
+        expected = files.lock_file(AGENT_NAME, self._config)
+        self.assertEqual(self._daemon.lockfile, expected)
 
     def test__daemonize(self):
         """Testing function _daemonize."""
         pass
 
     def test_delpid(self):
         """Testing function delpid."""
-        pass
+
+        # Creating daemon Process ID file(pidfile)
+        os.mknod(self._daemon.pidfile)
+
+        # Checking that pid file has been created
+        self.assertTrue(os.path.exists(self._daemon.pidfile))
+
+        # Delete pid file
+        self._daemon.delpid()
+
+        # Check that pidfile of the daemon has been deleted
+        self.assertFalse(os.path.exists(self._daemon.pidfile))
 
     def test_dellock(self):
         """Testing function dellock."""
-        pass
```

Comment:
Remove all the unnecessary pass statements. They were put in there just as placeholders until meaningful code was applied.